### PR TITLE
DOCS: Create Consolidated Troubleshooting Guide and Replace FAQ

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

- Removed FAQ.html (There were no FAQ only a couple troubleshooting items).
- Created troubleshooting.html.
  -  Moved troubleshooting content from old FAQ page here.
  -  Included troubleshooting from Partial files (in that form because they are non-contextual
  -  Added troubleshooting from specific pages that could be de-contextualized or are of a important enough context to warrant.

## Related issues

Resolves https://github.com/pomerium/docs/issues/57

## Checklist

- [x] reference any related issues
- [x] updated docs
- [ ] updated unit tests
- [ ] updated UPGRADING.md
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [ ] ready for review

### CODE REVIEW

File: docs/docs/troubleshooting.md
Comment (desimone) [inline docs/docs/troubleshooting.md:9]:
```suggestion
    content: pomerium troubleshooting faq frequently asked questions
```

File: docs/docs/troubleshooting.md
Comment (desimone) [inline docs/docs/troubleshooting.md:14]:
```suggestion
This article provides troubleshooting information for various tools and features in Pomerium.
```

defer to @cmo-pomerium tho

**Comment by alexfornuto** [comment]
Depending on which PR is merged first, either this or #2788 will want to add [this section](https://github.com/pomerium/pomerium/pull/2788/files#diff-70e9de814a62daf00d6df168b376adbc23e7cabb2a371084e0da1e58291c6a89R1-R7) to this page.

File: docs/docs/troubleshooting.md
Comment (desimone) [inline docs/docs/troubleshooting.md:45]:
```suggestion
Edit the upstream service configuration to use the local key to verify tokens.
```

File: docs/docs/troubleshooting.md
Comment (desimone) [inline docs/docs/troubleshooting.md:74]: - We can pull up HSTS to a higher level vs just being k8s based
- Reference the full header name here too `Strict-Transport-Security` too for search ability
- We should probably update the reference docs where this is set (https://www.pomerium.io/reference/#set-response-headers) to mention and link to (I think Mozilla does a really nice job here) each of the default security headers we set by default. (e.g. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security)

File: docs/docs/troubleshooting.md
Comment (desimone) [inline docs/docs/troubleshooting.md:22]: - Can we provide some additional context here around the different certificates at play? (JWKS / signing keys VS the lets encrypt certificates )
-  Maybe a little more explanation on why the LetsEncrypt issue results in the upstream client not being able to pull down and cache the jwks keys

**Review by desimone — CHANGES_REQUESTED** [review]

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md]: Adjusted in https://github.com/pomerium/pomerium/pull/2797/commits/414d1cd70d80d03ec22f6947f31da44bb36530db

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md]: Troubleshooting page adjuster in https://github.com/pomerium/pomerium/pull/2797/commits/58101e1bda4b42e22b6d944e8d0da8f9b577a6bb
Reference page adjusted in https://github.com/pomerium/pomerium/pull/2797/commits/6cc4df64fb5272d0804e3aafe5ea0049e5c1c590

**Review by desimone — APPROVED** [review]

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:38]:
```suggestion
This is often due to the recent expiration of the [DST Root CA X3](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021/) certificate. Many default keystores used by docker images and less-frequently updated distributions still carry this expired certificate. Even though Let's Encrypt (LE) certs are cross-signed with the [ISRG Root X1](https://letsencrypt.org/certificates/) CA certificate, some applications will still reject them.
```

If we don't re-insert Let's Encrypt in the above, let's define it before using the LE acronym.

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:40]:
```suggestion
To clarify; this does not mean that the upstream service is rejecting the JWT signing key. Rather, it doesn't trust the Let's Encrypt certificate used by the Authorize service for TLS, and so it will not read the JWKS file.
```
Assuming it's signing, not singing. Disregard if this is some specific jargon I don't know.

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:101]:
```suggestion
When using Redis, the [shared secret](/reference/readme.md#shared-secret) is used to encrypt data in Redis itself. If you change the configured shared secret, data from Redis can no longer be decrypted. This results in errant behavior, including redirect loops when a user session cannot be retrieved from the databroker.
```

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:107]: Should this all be in lower case? Not sure, just pointing it out

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:115]:
```suggestion
**Why You Are Seeing This**
```

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:117]:
```suggestion
This error means that the proxy is rejecting the authorized service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it. This can happen for a few reasons.
```
Regarding "This can happen for a few reasons" I propose explaining at least some of those reasons or just cutting it out.

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:147]:
```suggestion
**Why You Are Seeing This**
```

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:153]:
```suggestion
Usually, this is the result of either a routing issue or a configuration error. Make sure that you are using the _internally_ routable URL for authorizing the service. Many cloud loud balancers _do not_ yet support gRPC transposing the ingress. So while your authenticate service url will probably look like `https://authenticate.corp.example.com`, your authorizer service url will likely be more like `https://pomerium-authorize-service.default.svc.cluster.local` or `https://localhost:5443`.
```
Should "ingress" be capitalized in this?

File: docs/docs/troubleshooting.md
Comment (cmo-pomerium) [inline docs/docs/troubleshooting.md:165]:
```suggestion
When using Let's Encrypt certificates, you must use the `fullchain.pem` file, not `cert.pem` in order to include intermediate certs. Browsers like Chrome will store intermediate certs for LE but other tools (like `curl`) don't, which is why your route might look fine in a web browser, but not when curl'd or used for TCP tunneling.
```
Added "will" because my brain conjugated "Chrome store" as one noun and I had to re-read it again, so I doubt I'm the only one.

**Review by cmo-pomerium — COMMENTED** [review]
Finished copy review. Proposed some changes and made some comments. Please review and ask me if there's any questions.

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md:24]: I'm not a fan of the "as yet" tense in a technical doc, and I do want to cite examples of untrusted certs. How about:

```suggestion
When you visit an endpoint while Pomerium is using an untrusted certificate (like the self-signed bootstrap certificate or a Let's Encrypt staging certificate), that certificate may be pinned in your browser. Once Pomerium is switched to a trusted production certificate, the untrusted cert must reset in the browser.
```

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md:107]: Yes, because in this case the header is a direct copy of an error message. This should only occur in H4s, and when it does we copy the case of the error message for optimum search.

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md:115]: This makes sense in this instance, but thinking ahead: as we create more troubleshooting entries related to specific errors and/or log output, I want to have a standard format. "Why are you seeing this" will not always make sense in each context, so the more generic "why" is preferred.

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md:117]: :+1:  on reducing to "signed it", but the "authorize service" is correct. Pomerium is broken down into components: authorize, proxy, and the data broker. So in this case this isn't an adjective but a noun. It would be more obvious if it were a proper noun "Authorize service".... CC @desimone to validate that idea.

```suggestion
This error means that the proxy is rejecting the Authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it. This can happen for a few reasons.
```

> I propose explaining at least some of those reasons

I feel like this concern is addressed by the **Solution** section, which covers possible resolutions for several possible reasons. But if we don't have consensus on that idea, then cutting it out works for me.

```suggestion
This error means that the proxy is rejecting the authorize service's supplied certificate (used to establish a secure connection) because it doesn't know or trust the certificate authority that signed it.
```

File: docs/docs/troubleshooting.md
Comment (alexfornuto) [reply docs/docs/troubleshooting.md:153]: As discussed above, authorize is a noun not an adjective in this context. But the grammar was off either way. I copied this from the old FAQ page and did not review/rewrite it.
```suggestion
Usually, this is the result of either a routing issue or a configuration error. Make sure that you are using the _internally_ routable URL for the authorize service. Many cloud loud balancers _do not_ yet support gRPC transposing the ingress. So while your authenticate service url will probably look like `https://authenticate.corp.example.com`, your authorizer service url will likely be more like `https://pomerium-authorize-service.default.svc.cluster.local` or `https://localhost:5443`.
```

re "ingress": in this context we're not explicitly talking about a Kubernetes object type, so we're not bound by their... questionable choices in capitalization.

**Comment by alexfornuto** [comment]
I'll merge review suggestions after hearing back on the idea to capitalize Pomerium service names.

**Pull request merged**
